### PR TITLE
ipynb repr

### LINKED ISSUE #219: Create a Garden or Pipeline preview in notebooks using `_repr_html_()`
_In a similar style to Foundry datasets, define a `repr_html` function to nicely display the Garden or Pipeline metadata_

Ben: Adding a notebook-specific preview of the Garden would be a nice student task. Something that gives a nice visual repr like this. To do that, you just define a repr_html function and notebooks look for that automagically. 

image (10)

See [the Foundry code](https://github.com/MLMI2-CSSI/foundry/blob/35c2e12418660cf6ffe6995dbc446dca7183250a/foundry/foundry.py#L330) for an example

# Assumptions:
1.
2.

# Acceptance Criteria
Given..., when..., then...

### PR DESCRIPTION
**Pull request by isaac-darling**
`Fixes #219`

## Overview

Implement notebook `repr` functions to pretty print gardens and pipelines.

## Testing

I rendered examples in notebooks, and they look as expected. (screenshots included)

----
:books: Documentation preview :books:: https://garden-ai--234.org.readthedocs.build/en/234/

### CODE REVIEW

**Comment by isaac-darling** [comment]
garden_repr
pipeline_repr

**Review by WillEngler — COMMENTED** [review]
Overall approach looks good and the sample output looks good. Asking for a second pass on the pipeline implementation. Also looks like there is some type checking issue causing CI to fail.

File: garden_ai/pipelines.py [hunk 1]
```diff
@@ -528,3 +535,42 @@ def expanded_metadata(self) -> Dict[str, Any]:
         data = self.dict()
         data["models"] = [m.dict() for m in self.collect_models()]
         return data
+
+
+def pipeline_repr_html(pipeline: Union[Pipeline, RegisteredPipeline]) -> str:
+    def prettify_type_repr(s: str) -> str:
+        if "<" in s:
+            return s[s.find("<class '") + 8 : s.rfind("'")]
+        return s
```
Comment (WillEngler) [inline garden_ai/pipelines.py:544]: Nice, breaking this little function out makes a big difference

**Review by WillEngler — APPROVED** [review]
LGTM, this is going to be a great add to the notebook experience. Thanks Isaac!

**Pull request merged**